js/components/thermoscope.js

```diff
@@ -36,21 +40,32 @@ export default class Thermoscope extends PureComponent {
     this.setState({materialIdx: value});
   }
 
+  liveDataHandler() {
+    if (this.sensor && this.sensor.liveSensors) {
+      let newData = this.sensor.liveSensors[this.props.probeIndex].liveValue;
+      if (!isNaN(newData) && isFinite(newData)) this.setState({ temperature: newData, liveData: true });
+    }
+  }
+
   render() {
-    const { temperature, materialType, materialIdx } = this.state;
+    const { temperature, materialType, materialIdx, liveData } = this.state;
     const model = models[materialType][materialIdx];
+    let slider;
+    if (!liveData){
+    slider = <Slider min={MIN_TEMP} max={MAX_TEMP} step={1} value={temperature}
+        sliderStyle={{ marginTop: 5, marginBottom: 5 }}
+        name="temperature"
+        onChange={this.handleTempSliderChange} />
+    }
+
     return (
       <div className="thermoscope">
-        <LabModel model={model.json} tempScale={model.tempScale} temperature={temperature}  width={MODEL_WIDTH} height={MODEL_HEIGHT}/>
-
+        <LabModel model={model.json} tempScale={model.tempScale} temperature={temperature} liveData={liveData} width={MODEL_WIDTH} height={MODEL_HEIGHT}/>
         <div>
           <div className="controls-row">
             Temperature {temperature}°C
             <div className="slider">
-              <Slider min={MIN_TEMP} max={MAX_TEMP} step={1} value={temperature}
-                      sliderStyle={{marginTop: 5, marginBottom: 5}}
-                      name="temperature"
-                      onChange={this.handleTempSliderChange}/>
+              {slider}
```

Comment:
A matter of taste, but personally I like to use following pattern:
```javascript
{!liveData && <Slider min={MIN_TEMP} max={MAX_TEMP} .../>}
```
so it's visible that slider is displayed conditionally. But this one works too, either way is fine.